src/rpc/daemon_messages.cpp

```diff
@@ -0,0 +1,989 @@
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "daemon_messages.h"
+#include "serialization/json_object.h"
+
+namespace cryptonote
+{
+
+namespace rpc
+{
+
+const char* GetHeight::name = "get_height";
+const char* GetBlocksFast::name = "get_blocks_fast";
+const char* GetHashesFast::name = "get_hashes_fast";
+const char* GetTransactions::name = "get_transactions";
+const char* KeyImagesSpent::name = "key_images_spent";
+const char* GetTxGlobalOutputIndices::name = "get_tx_global_output_indices";
+const char* GetRandomOutputsForAmounts::name = "get_random_outputs_for_amounts";
+const char* SendRawTx::name = "send_raw_tx";
+const char* GetInfo::name = "get_info";
+const char* SaveBC::name = "save_bc";
+const char* GetBlockHash::name = "get_block_hash";
+const char* GetLastBlockHeader::name = "get_last_block_header";
+const char* GetBlockHeaderByHash::name = "get_block_header_by_hash";
+const char* GetBlockHeaderByHeight::name = "get_block_header_by_height";
+const char* GetPeerList::name = "get_peer_list";
+const char* SetLogLevel::name = "set_log_level";
+const char* GetTransactionPool::name = "get_transaction_pool";
+const char* HardForkInfo::name = "hard_fork_info";
+const char* GetOutputHistogram::name = "get_output_histogram";
+const char* GetOutputKeys::name = "get_output_keys";
+const char* GetRPCVersion::name = "get_rpc_version";
+
+
+
+
+rapidjson::Value GetHeight::Request::toJson(rapidjson::Document& doc)
+{
+  return Message::toJson(doc);
+}
+
+void GetHeight::Request::fromJson(rapidjson::Value& val)
+{
+}
```

Comment:
hmm...technically can be empty, as there are no meaningful fields to deserialize, but I could have it call the parent class' fromJson for posterity.  I'll ponder it.